src/welcome/ProgressiveProfiling.jsx

```diff
@@ -69,6 +69,13 @@ const ProgressiveProfiling = (props) => {
     return null;
   }
 
+  if (props.location.state && Object.keys(props.location.state.optionalFields.fields).length > 0) {
+    optionalFields = props.location.state.optionalFields.fields;
+  }
+
+  if (props.location.state && Object.keys(props.location.state.optionalFields.extended_profile).length > 0) {
+    extendedProfile = props.location.state.optionalFields.extended_profile;
+  }
```

Comment:
You don't need to check props.location.state, it is already checked [here](https://github.com/openedx/frontend-app-authn/pull/716/files#diff-938bc6bf0a58a5670564197db26d51affa26ba8224d1bb732d321d3145249b1cR63).

Also you should not proceed if optional fields are not present.
```suggestion
  if (!props.location.state.optionalFields) {
    global.location.assign(DASHBOARD_URL);
    return null;
  }
  
  const optionalFields = props.location.state.optionalFields.fields;
  const extendedProfile = props.location.state.optionalFields.extended_profile;
```